pkg/datapath/iptables/iptables.go

```diff
@@ -168,6 +168,21 @@ func (m *IptablesManager) removeCiliumRules(table, prog, match string) {
 		// -A CILIUM_FORWARD -o cilium_host -m comment --comment "cilium: any->cluster on cilium_host forward accept" -j ACCEPT
 		// -A POSTROUTING -m comment --comment "cilium-feeder: CILIUM_POST" -j CILIUM_POST
 		if strings.Contains(rule, match) {
+			// do not remove feeder for chains that are set to be disabled
```

Comment:
Is this point just for feeders or is it for all of the rules? The surrounding code seems to suggest all rules.

Reply:
Well it's for feeders since they are the only ones in preexisting chains. That's why I have to add a space before and after on the check to get the part when the chain is specified. Not sure if I was clear 😅

Reply:
What I mean is that the top of this function just does something like `iptables -t filter -S`, which lists all rules in the table, both feeder and non-feeder rules. The `strings.Contains()` is really just matching rules that contain the string `CILIUM_`. So it's not just the feeder rules from the main chains (eg FORWARD), but also the rules inside the cilium-specific chains.

When this new code filters these chains further, it's really just looking for anything with the string something like `CILIUM_FORWARD` in it, which according to the lines 168-169 directly above here, could either match on a feeder rule that has a `... -j CILIUM_FORWARD` type jump, or also it could match rules inside the `CILIUM_FORWARD` chain like `-A CILIUM_FORWARD ...`.

The result of this would be that when you use this commandline option, it doesn't just skip the feeder rule but other rules. So the rest of the Cilium chain might be out-of-date.

Reply:
Hmm the thing is when I tried without this block my rules that I inserted (eg `iptables -A POSTROUTING m comment --comment "cilium-feeder: CILIUM_POST" -j CILIUM_POST`) was deleted as well. 
When reading the code it's seems legit since it matchs the `CILIUM_` string in the chain at the end.
So the goal of the code was to skip all rules which looks like `-A/I <main_chain>` where main chain is in the new flag.
And if I understand it correctly it will only match these feeder rules 🤔

Reply:
Ah, thanks for pointing it out.

Feeder rule for postrouting:
```
-A POSTROUTING ... -j CILIUM_POST
```

So in this case if you specify `match` to `POSTROUTING` it will only match the feeder rule. The regular rules have `-A CILIUM_POST` which do not match `POSTROUTING`.

Feeder rule for FORWARD:
```
-A FORWARD -j CILIUM_FORWARD
```

Unless I'm missing something, if you specified `FORWARD`, it will match all rules for the feeder (like above) and all of the `-A CILIUM_FORWARD` rules as well, because this is just doing a substring match.

Maybe this could be solved as simply as just ensuring `match` has a space before and after. That way we know that the whole word, eg ` FORWARD ` is present and is not a subset of `CILIUM_FORWARD` for the rule?

Reply:
@joestringer the spaces are already there :smile:

Reply:
:man_facepalming: I don't know what I was looking at.